borsh/src/de/mod.rs

```diff
@@ -91,15 +92,17 @@ impl BorshDeserialize for u8 {
 
     #[inline]
     #[doc(hidden)]
-    fn copy_from_bytes(buf: &mut &[u8], out: &mut [Self]) -> Result<bool> {
+    fn copy_from_bytes(buf: &mut &[u8], out: &mut [MaybeUninit<Self>]) -> Result<bool> {
```

Comment:
This technically makes the crate `unsafe`: if you override this method, but do not init all values, the `assume_init_read` below becomes unsound.